src/Controls/src/Core/Application.cs

```diff
@@ -19,8 +20,6 @@ public partial class Application : Element, IResourcesProvider, IApplicationCont
 		readonly Lazy<PlatformConfigurationRegistry<Application>> _platformConfigurationRegistry;
 		readonly Lazy<IResourceDictionary> _systemResources;
 
-		public override IDispatcher Dispatcher => this.GetDispatcher();
```

Comment:
that is the issue... Application does not have a UI thread component. My initial case was to say that all bindable objects need a UI thread - like all the other platforms do, but unfortunately this breaks a lot of tests and things. We can change it later and just throw in the ctor if not on the UI thread, but that is for another PR.

However, right now if you create an object on a thread with a dispatcher, that is the thread you must stay on. If you do a background thread, then it will try find it each time and look up the UI hierarchy or fall back to Application.

It is wrong, but this is what forms did.

The correct solution is to not allow UI objects to be created off the UI thread. This should throw if null is returned: https://github.com/dotnet/maui/pull/2862/files#diff-ebb14ec14b8baeffdd7b0ae40ffc8bf0e8944f76fdea50fcbc737fb142bfb44aR24

I want to do that later. However, this will break people that create objects of the UI thread.

Reply:
Would it be useful to have it throw if `Windows.Count > 1`?

Reply:
IMO, `Application.Current.Dispatcher` is similar to `Device.BeginInvokeOnMainThread()`: They both assume that the app can only ever have a single window.
This is fine (and convenient) for the majority of apps, but it is also a trap for the minority of developers who build multi-window apps. The worst scenario is when that developer takes a dependency on a library that uses these APIs.

Based on that, I think that:
- This API should never be used by libraries (including MAUI itself and the toolkit). An idea would even be to put something like `#if DEBUG`, add the `[Obsolete]` attribute to at least warn MAUI devs.
- We should move this singleton `Dispatcher` to a place that has a name that says "Use this only if you are sure that this code will only be used in single window apps" (something more explicit than a comment in the documentation that most devs will not read).

Reply:
The singleton Dispatcher does not actually provide a dispatcher. It just gets the one for the current thread - like WPF/WinUI. If there is no dispatcher for the thread, it returns null. That is why I renamed the method to `Dispatcher.GetForCurrentThread()` to make it very obvious.

For Application.Dispatcher, that property can't go away as it is on the base type, however, this is a weird one because the app actually exists on a thread with a dispatcher. When you start an app, there is this "main thread" or "initial thread" that contains the app and first window. This has a dispatcher. When you close the window, the thread is still there.

So, maybe we should throw if there are multiple windows, however, this is also not true. I can have an app on a thread without windows, and that app has a dispatcher. And, I have have an app with multiple windows on the same thread.

If you do Application.Dispatcher, this is a valid state. However, doing that is not wise because you do not know _which_ window it runs on. It may not even be on _any_ window. The dispatcher is still valuable on Application because you may want to coordinate operations in a sequence on Application. For example, setting some properties. We have no properties right now at a core level, but there might be at some point. For example, maybe there is a set of properties to set some app theme. Maybe for thread safety, this has to be set on the same thread that the app is running on (just like with WinUI) so we need to use the Application.Dispatcher. There might be no window on that thread - and that is fine.

This is why it is a _dispatcher_ and not a _main thread_. This is the main reason I prefer this term as opposed to "run on MAIN thread" or "run on UI thread".

Reply:
The real issue here is that I can create a Button, for example, on some background thread and try use the Dispatcher. This is invalid to be honest, but this is what might be happening today. This PR can't really solve that as I want to open a new PR with that change and start a discussion on this. I am moving towards this state right now by trying to find a dispatcher and throwing if one can't be found.

I am hoping to turn on the stricter creating as there is no real advantage on creating UI elements on a background thread. If your UI is so complex that it takes seconds to create, imagine how long it will take to create all the native elements. Maui UI elements are pretty basic and do not really do any platform things. If your UI takes 10 seconds to create in Maui, then it will be 10 minutes in native Android.

This PR moves the dispatchers to Core and tries to preserve existing behavior. The next one will do it right, or at least attempt to convince the world to do it right.